[DebugInfo] Correct an overly-restrictive REQUIRES clause.

CarlosAlbertoEnciso opened the pull request:
Include a regular expression in the 'REQUIRES' clause, to run
the test on all matching targets (x86_64 *linux*).

The original patch restricted to test just to 'x86_64-linux'
https://github.com/llvm/llvm-project/pull/116327

CarlosAlbertoEnciso committed:
[DebugInfo] Add DW_AT_artificial for compiler generated static member. (#116327)

Consider the case when the compiler generates a static member. Any
consumer of the debug info generated for that case, would benefit if
that member has the DW_AT_artificial flag.

Fix the syntax for 'REQUIRES', to include a regular expression.

jmorse reviewed (approved):
LGTM, but you should write the commit message to be directly about what you're changing here, not carry with it the context of what the original patch did. Someone viewing this in isolation could be confused by that -- better to instead title it "Correct an overly-restrictive REQUIRES clause", another sentence of explanation, and then a link back to the original patch this landed in in case someone wants to read up on it.

CarlosAlbertoEnciso committed:
[DebugInfo] Correct an overly-restrictive REQUIRES clause.

Include a regular expression in the 'REQUIRES' clause, to run
the test on all matching targets (x86_64 *linux*).

The original patch restricted to test just to 'x86_64-linux'
https://github.com/llvm/llvm-project/pull/116327

CarlosAlbertoEnciso commented:
@jmorse, @dwblaikie Thanks for your feedback.
Looking into more detail, the regular expression should be: `{{x86_64-.*-linux.*}}`
Updated title and commit comment.
